src/Compilers/Core/Desktop/AssemblyUtilities.cs

```diff
@@ -0,0 +1,188 @@
+        /// reference against a file name; if they match the file is considered a
+        /// dependency. Other factors, such as version, culture, public key, etc., 
+        /// are not considered, and so the returned collection may include items that
+        /// cannot in fact satisfy the original assembly's dependencies.
+        /// </remarks>
+        /// <exception cref="IOException">If the file at <paramref name="filePath"/> does not exist or cannot be accessed.</exception>
+        /// <exception cref="BadImageFormatException">If the file is not an assembly or is somehow corrupted.</exception>
+        public static ImmutableArray<string> FindAssemblySet(string filePath)
+        {
+            Debug.Assert(filePath != null);
+            Debug.Assert(PathUtilities.IsAbsolute(filePath));
+
+            Queue<string> workList = new Queue<string>();
+            HashSet<string> assemblySet = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+
+            workList.Enqueue(filePath);
+
+            while (workList.Count > 0)
+            {
+                string assemblyPath = workList.Dequeue();
+
+                assemblySet.Add(assemblyPath);
+
+                var directory = Path.GetDirectoryName(assemblyPath);
+
+                using (var reader = new PEReader(FileUtilities.OpenRead(assemblyPath)))
+                {
+                    var metadataReader = reader.GetMetadataReader();
+                    var assemblyReferenceHandles = metadataReader.AssemblyReferences;
+
+                    foreach (var handle in assemblyReferenceHandles)
+                    {
+                        var reference = metadataReader.GetAssemblyReference(handle);
+                        var referenceName = metadataReader.GetString(reference.Name);
+
+                        string referencePath = Path.Combine(directory, referenceName + ".dll");
+
+                        if (!assemblySet.Contains(referencePath) &&
+                            File.Exists(referencePath))
+                        {
+                            workList.Enqueue(referencePath);
+                        }
+                    }
+                }
+            }
+
+            return ImmutableArray.CreateRange(assemblySet);
+        }
+
+        /// <summary>
+        /// Given a path to an assembly, returns its MVID (Module Version ID).
+        /// May throw.
+        /// </summary>
+        /// <exception cref="IOException">If the file at <paramref name="filePath"/> does not exist or cannot be accessed.</exception>
+        /// <exception cref="BadImageFormatException">If the file is not an assembly or is somehow corrupted.</exception>
+        public static Guid ReadMvid(string filePath)
+        {
+            Debug.Assert(filePath != null);
+            Debug.Assert(PathUtilities.IsAbsolute(filePath));
+
+            using (var reader = new PEReader(FileUtilities.OpenRead(filePath)))
```

Comment:
@heejaechang No, it's too specific. You could have two different analyzers with the same full assembly name; I doubt you would want to run both.